src/darwin/Framework/CHIP/templates/MTRClusters-src.zapt

```diff
@@ -52,56 +53,71 @@ using chip::SessionHandle;
 {
     // Make a copy of params before we go async.
     params = [params copy];
-    MTRBaseDevice *baseDevice = [[MTRBaseDevice alloc] initWithNodeID:self.device.nodeID controller:self.device.deviceController];
-    new MTR{{>callbackName}}CallbackBridge(self.callbackQueue,
-      baseDevice,
-      {{#if hasSpecificResponse}}
-        {{! This treats completionHandler as taking an id for the data.  This is
-            not great from a type-safety perspective, of course. }}
-        completionHandler,
-      {{else}}
-        {{! For now, don't change the bridge API; instead just use an adapter
-            to invoke our completion handler. This is not great from a
-            type-safety perspective, of course. }}
-        ^(id _Nullable value, NSError * _Nullable error) {
-          completionHandler(error);
-        },
-      {{/if}}
-      ^(ExchangeManager & exchangeManager, const SessionHandle & session, Cancelable * success, Cancelable * failure) {
-        chip::Optional<uint16_t> timedInvokeTimeoutMs;
-        ListFreer listFreer;
-        {{asUpperCamelCase parent.name}}::Commands::{{asUpperCamelCase name}}::Type request;
-        if (params != nil) {
-          if (params.timedInvokeTimeoutMs != nil) {
-            timedInvokeTimeoutMs.SetValue(params.timedInvokeTimeoutMs.unsignedShortValue);
-          }
-        }
-        {{#if mustUseTimedInvoke}}
-        if (!timedInvokeTimeoutMs.HasValue()) {
-          timedInvokeTimeoutMs.SetValue(10000);
-        }
-        {{/if}}
-        {{#chip_cluster_command_arguments}}
-          {{#first}}
-            {{#unless (commandHasRequiredField parent)}}
-            if (params != nil) {
-            {{/unless}}
-          {{/first}}
-          {{>encode_value target=(concat "request." (asLowerCamelCase label)) source=(concat "params." (asStructPropertyName label)) cluster=parent.parent.name errorCode="return CHIP_ERROR_INVALID_ARGUMENT;" depth=0}}
-          {{#last}}
-            {{#unless (commandHasRequiredField parent)}}
+    NSNumber *timedInvokeTimeoutMsParam = params.timedInvokeTimeoutMs;
```

Comment:
What will this return if `params` is nil?

Reply:
Objc is convenient in that there's default value for nil. In this case timedInvokeTimeoutMsParam will be set to nil, which is correct.